### PR TITLE
[SPARK-22350][SQL] select grouping__id from subquery

### PR DESCRIPTION
**Pull request by jinxing64**
## What changes were proposed in this pull request?

Currently, sql below will fail:
```
SELECT cnt, k2, k3, grouping__id
FROM
(SELECT count as cnt, k2, k3, grouping__id
FROM t1
GROUP BY k2, k3
GROUPING SETS(k2, k3)) t2
```
The use case is common in our warehouse and supported by hive now. 
Could we support it?

## How was this patch tested?
Test added

### CODE REVIEW

**Comment by DonnyZone** [comment]
Is it similar to the below issue?
https://github.com/apache/spark/pull/19178

**Comment by jinxing64** [comment]
@DonnyZone 
Thanks for taking a look.
I think not quite the same.
After https://github.com/apache/spark/pull/18270, all `grouping__id` are transformed to be `GroupingID` , which makes user cannot select `grouping__id` with subquery.
Also after that pr, `grouping__id` is not deprecated any longer. This pr removes `spark_grouping_id` and simplify the logic.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala [hunk 1]
```diff
@@ -1497,6 +1497,27 @@ class SQLQuerySuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
     }
   }
 
+  test("select grouping__id from subquery.") {
+    checkAnswer(
+      sql(
+        """
+          |SELECT cnt, k2, k3, grouping__id
+          |FROM
+          |  (SELECT count(*) as cnt, k2, k3, grouping__id
```
Comment (gatorsmile) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/SQLQuerySuite.scala]:
```
SELECT cnt, k2, k3, alias_grouping__id
FROM
  (SELECT count(*) as cnt, k2, k3, grouping__id as alias_grouping__id
  FROM (SELECT key, key%2 as k2 , key%3 as k3 FROM src) t1
  GROUP BY k2, k3
  GROUPING SETS(k2, k3)) t2
ORDER BY alias_grouping__id, k2, k3
```
This is the workaround.

**Comment by jinxing64** [comment]
@gatorsmile
thanks for reply.
It seems you preffer to give the alias explicitly. I will close this pr and go by your suggestion.
But in my warehouse, there are lots of ETLs which are selecting grouping__id from subquery. We cannot migrate seamlessly

**Comment by gatorsmile** [comment]
@jinxing64 You can keep it open, but it might take more time to review the fix.

**Pull request closed without merging**